dubbo-rpc/dubbo-rpc-rest/src/test/java/org/apache/dubbo/rpc/protocol/rest/RestProtocolTest.java

```diff
@@ -49,16 +50,55 @@ public void tearDown() {
         protocol.destroy();
     }
 
+    @Test
+    public void testRestProtocol() {
+        URL url = URL.valueOf("rest://127.0.0.1:5342/rest/say?version=1.0.0");
+        DemoServiceImpl server = new DemoServiceImpl();
+        ProviderModel providerModel = new ProviderModel(url.getServiceKey(), server, DemoService.class);
+        ApplicationModel.initProviderModel(url.getServiceKey(), providerModel);
+
+        Exporter<DemoService> exporter = protocol.export(proxy.getInvoker(server, DemoService.class, url));
+        Invoker<DemoService> invoker = protocol.refer(DemoService.class, url);
+        Assertions.assertFalse(server.isCalled());
+
```

Comment:
would you care to split the UT in more logic unit e.g.

```
@Test
public void testServerIsCalled() {
   URL url = URL.valueOf("rest://127.0.0.1:5342/rest/say?version=1.0.0");
        DemoServiceImpl server = new DemoServiceImpl();
        ProviderModel providerModel = new ProviderModel(url.getServiceKey(), server, DemoService.class);
        ApplicationModel.initProviderModel(url.getServiceKey(), providerModel);

        Exporter<DemoService> exporter = protocol.export(proxy.getInvoker(server, DemoService.class, url));
        Invoker<DemoService> invoker = protocol.refer(DemoService.class, url);
}
```

```
@Test
public void testRestProtocolReturnResult() {
   URL url = URL.valueOf("rest://127.0.0.1:5342/rest/say?version=1.0.0");
        DemoServiceImpl server = new DemoServiceImpl();
        ProviderModel providerModel = new ProviderModel(url.getServiceKey(), server, DemoService.class);
        ApplicationModel.initProviderModel(url.getServiceKey(), providerModel);

        Exporter<DemoService> exporter = protocol.export(proxy.getInvoker(server, DemoService.class, url));
        Invoker<DemoService> invoker = protocol.refer(DemoService.class, url);
       DemoService client = proxy.getProxy(invoker);
        String result = client.sayHello("haha");
        Assertions.assertTrue(server.isCalled());
        Assertions.assertEquals("Hello, haha", result);
        invoker.destroy();
        exporter.unexport();
}
```

To me in the suggested way, if anything breaks, looking at UT we can get more closer to place where it has break. What do you say?

Reply:
I think the impact is not great. The testRestProtocol itself is to detect whether the Rest Protocol is successfully used.